### ISSUE TITLE
Number of input channels

### ISSUE THREAD

**Issue by gbalduzz** [ultralytics/ultralytics#22504]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Dear all,
I would like to train the yolov11 architecture on grayscale images. 
I followed this issue on yolov5 https://github.com/ultralytics/yolov5/issues/1739, but setting the number of channels in the model .yaml seems to no longer be supported, i.e., I added `ch: 1` to yolo11.yaml, but when loading it, I still get 3 input channels. 
```
                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2] 
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @gbalduzz, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you unblocked quickly—an Ultralytics engineer will also assist you here soon.

For general guidance, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, and where many common questions may already be answered.

Since your question involves using YOLO11 with grayscale images and modifying input channels, please share a few extra details to help us understand the setup:
- Your exact `model.yaml` snippet (where you attempted to set channels)
- The code snippet you used to load/initialize the model
- `ultralytics` and `torch` versions, plus OS/CPU/GPU details
- A brief log or console output showing the 3-channel input at model build time

If this is a 🐛 Bug Report (i.e., the configuration is ignored), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) including:
- Minimal code to reproduce from a clean environment
- The custom YAML you’re using
- Any logs/tracebacks
- One small grayscale sample image (if relevant)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). This will help us advise the best path for grayscale workflows 👍

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You need to set `channels` to 1 in your dataset YAML

https://docs.ultralytics.com/datasets/detect/coco8-multispectral/#dataset-yaml

**Comment by gbalduzz** [comment]
Many thanks!

**Comment by glenn-jocher** [comment]
Great—if you want a quick sanity check, train on the COCO8‑Grayscale example (it builds a 1‑channel model) and confirm layer 0 shows 1 input channel; see the [COCO8‑Grayscale dataset docs](https://docs.ultralytics.com/datasets/detect/coco8-grayscale/).

```bash
yolo detect train data=coco8-grayscale.yaml model=yolo11n.pt epochs=1 imgsz=640
```
